JitsiConference.js

```diff
@@ -1623,7 +1786,519 @@ JitsiConference.prototype.broadcastEndpointMessage = function(payload) {
+        () => {
+            logger.info(`Detach local tracks from JVB done!${localTracks}`);
+        },
+        error => {
+            logger.info(
+                `Detach local tracks from JVB failed!${localTracks}`, error);
+        });
+};
+
+/**
+ * Clears the deferred start P2P task if it has been scheduled.
+ * @private
+ */
+JitsiConference.prototype._maybeClearDeferredStartP2P = function() {
+    if (this.deferredStartP2P) {
+        logger.info('Cleared deferred start P2P task');
+        clearTimeout(this.deferredStartP2P);
+        this.deferredStartP2P = null;
+    }
+};
+
+/**
+ * Removes from the conference remote tracks associated with the JVB
+ * connection.
+ * @private
+ */
+JitsiConference.prototype._removeRemoteJVBTracks = function() {
+    this._removeRemoteTracks('JVB', this.jvbJingleSession);
+};
+
+/**
+ * Removes from the conference remote tracks associated with the P2P
+ * connection.
+ * @private
+ */
+JitsiConference.prototype._removeRemoteP2PTracks = function() {
+    this._removeRemoteTracks('P2P', this.p2pJingleSession);
+};
+
+/**
+ * Generates fake "remote track removed" events for given Jingle session.
+ * @param {string} sessionNickname the session's nickname which will appear in
+ * log messages.
+ * @param {JingleSessionPC} jingleSession the session for which remote tracks
+ * will be removed.
+ * @private
+ */
+JitsiConference.prototype._removeRemoteTracks
+= function(sessionNickname, jingleSession) {
+    if (!jingleSession) {
+        logger.info(
+            `Not removing remote ${sessionNickname} tracks - no session yet`);
+
+        return;
+    }
+    const remoteTracks = jingleSession.peerconnection.getRemoteTracks();
+
+    for (const track of remoteTracks) {
+        logger.info(`Removing remote ${sessionNickname} track: ${track}`);
+        this.rtc.eventEmitter.emit(RTCEvents.REMOTE_TRACK_REMOVED, track);
+    }
+};
+
+/**
+ * Sets new P2P status and updates some events/states hijacked from
+ * the <tt>JitsiConference</tt>.
+ * @param {boolean} newStatus the new P2P status value, <tt>true</tt> means that
+ * P2P is now in use, <tt>false</tt> means that the JVB connection is now in use
+ * @private
+ */
+JitsiConference.prototype._setP2PStatus = function(newStatus) {
+    if (this.p2pEstablished === newStatus) {
+        logger.error(`Called _setP2PStatus with the same status: ${newStatus}`);
+
+        return;
+    }
+    this.p2pEstablished = newStatus;
+    if (newStatus) {
+        logger.info('Peer to peer connection established!');
+    } else {
+        logger.info('Peer to peer connection closed!');
+    }
+
+    // Clear dtmfManager, so that it can be recreated with new connection
+    this.dtmfManager = null;
+
+    // Update P2P status
+    this.eventEmitter.emit(
+        JitsiConferenceEvents.P2P_STATUS,
+        this,
+        this.p2pEstablished);
+
+    // Refresh connection interrupted/restored
+    this.eventEmitter.emit(
```

Comment:
No, when we switch back to JVB connection we need to update to reflect JVB's connection status.